### DISCUSSION TITLE
Make easier to cross compilation of Zed

### DISCUSSION THREAD

**Discussion by yodatak** [zed-industries/zed#15648] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I run on linux and i want to compile zed for windows but its a pain 
we would use something like cross or use this https://bevy-cheatbook.github.io/setup/cross/linux-windows.html

but i cannot make it works we should make a documentation and permit to cross compilation make simple

**Comment by yodatak** [comment]
For now the more i can do is this 

```sh
git clone https://github.com/cross-rs/cross
cd cross
git submodule update --init --remote
cargo build-docker-image x86_64-pc-windows-msvc-cross --tag local
```

add file Cross.toml and add this into 
```
[target.x86_64-pc-windows-msvc]
image = "ghcr.io/cross-rs/x86_64-pc-windows-msvc-cross:local"
```

 remove the 2 linux targets `./cargo/config.toml`
 mold option because https://github.com/cross-rs/cross/issues/1268
```
# [target.x86_64-unknown-linux-gnu]
# [target.aarch64-unknown-linux-gnu]
```

```sh
cross build --target x86_64-pc-windows-msvc --release

  = note: 0074:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
          0074:err:winediag:nodrv_CreateWindow L"The explorer process failed to start."
          0074:err:systray:initialize_systray Could not create tray window
          0024:err:environ:build_command_line command line too long (84840)
          

The following warnings were emitted during compilation:

warning: zed@0.148.0: Info: using '4bfb8fda8d684bac1992502c26698ac734eb5827' hash for ZED_COMMIT_SHA env var

error: could not compile `zed` (bin "zed") due to 1 previous error

```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
